app/models/tag.rb

```diff
@@ -1023,7 +1033,7 @@ def remove_sub_filters(sub_tag)
   before_update :check_canonical
   def check_canonical
     if self.canonical_changed? && !self.canonical?
-      self.async(:remove_canonical_associations)
+      remove_canonical_associations
```

Comment:
They're only changing the associations, not the FilterTaggings or the associated works. (And those changes are generally made on the tag edit pages, which already have to load all of the associations in order to display them.) But I can change them back to async and try to modify the tests instead, if that would be preferable.